background_scripts/commands.js

```diff
@@ -260,21 +260,31 @@ const Commands = {
     const commandToKey = {};
     for (const key of Object.keys(this.keyToRegistryEntry || {})) {
       const registryEntry = this.keyToRegistryEntry[key];
-      (commandToKey[registryEntry.command] != null
-        ? commandToKey[registryEntry.command]
-        : (commandToKey[registryEntry.command] = [])).push(key);
+      const optionString = registryEntry.optionList?.length > 0
+        ? registryEntry.optionList?.join(" ")
+        : "";
+      if (commandToKey[registryEntry.command] == null) {
+        commandToKey[registryEntry.command] = {};
+      }
+      (commandToKey[registryEntry.command][optionString] != null
+        ? commandToKey[registryEntry.command][optionString]
+        : (commandToKey[registryEntry.command][optionString] = [])).push(key);
     }
     const commandGroups = {};
     for (const group of Object.keys(this.commandGroups || {})) {
       const commands = this.commandGroups[group];
       commandGroups[group] = [];
       for (const command of commands) {
-        commandGroups[group].push({
-          command,
-          description: this.availableCommands[command].description,
-          keys: commandToKey[command] != null ? commandToKey[command] : [],
-          advanced: this.advancedCommands.includes(command),
-        });
+        for (const [options, keys] of Object.entries(commandToKey[command] ?? { "": [] })) {
+          commandGroups[group].push({
+            command,
+            description: this.availableCommands[command].description,
+            keys: keys,
+            advanced: this.advancedCommands.includes(command) ||
+              this.advancedCommands.includes(`${command} ${options}`),
```

Comment:
This is the line for allowing command + options sets to be considered advanced (like `reload hard`). We just also check if the advancedCommands list includes the command + options set.